memote/support/biomass.py

```diff
@@ -176,14 +192,83 @@ def find_direct_metabolites(model, reaction):
         that are taken up to be consumed by the biomass reaction only.
 
     """
-    transport_reactions = set(helpers.find_transport_reactions(model))
-    exchange_reactions = set(model.exchanges)
-    biomass_reactions = set(helpers.find_biomass_reaction(model))
-
-    combined_set = transport_reactions | exchange_reactions | biomass_reactions
-    precursors = find_biomass_precursors(model, reaction)
-
-    return [met for met in precursors if met.reactions.issubset(combined_set)]
+    biomass_rxns = set(helpers.find_biomass_reaction(model))
+    tra_bou_bio_rxns = helpers.find_tra_bou_bio_reactions(model, biomass_rxns)
+    try:
+        precursors = find_biomass_precursors(model, reaction)
+        main_comp = helpers.find_compartment_id_in_model(model, 'c')
+        ext_space = helpers.find_compartment_id_in_model(model, 'e')
+    except KeyError as k:
```

Comment:
Other than in list comprehensions single letter variables are bad style. It'd be better to use `as err` or even `as error` here.

Reply:
Got it! I'll fix this in the next commit!